addon/src/main/java/fi/jasoft/dragdroplayouts/DragCaption.java

```diff
@@ -0,0 +1,32 @@
+package fi.jasoft.dragdroplayouts;
+
+import com.vaadin.server.Resource;
+
+/**
+ * Drag caption definition: icon and caption.
+ */
+public class DragCaption {
+    protected String caption;
+    protected Resource icon;
+
+    public DragCaption(String caption, Resource icon) {
```

Comment:
Can the caption contain HTML? If yes, should filter through jsoup or explicitly document the that it does (potential XSS vulnerability). In core we have recently provided the html mode with an extended API.